backend/oasst_backend/user_stats_repository.py

```diff
@@ -227,6 +227,9 @@ def get_trollboard(
         if enabled is not None:
             qry = qry.filter(User.enabled == enabled)
 
+        if lang is not None and not "":
+            qry = qry.filter(User.most_used_lang(session=self.session) == lang)
```

Comment:
This calls the function `User.most_used_lang()` (a classmethod) with the first parameter being the class-object .. e.g. if you internally in the function refer to self.id it should give you the Field() value. Overall this function will be executed once for the whole query independent of how many users there are on the trollboard. I currently don't see how/why this could work. Did you really test this with multiple users with different languages and you see if you have 10 users on the trollboard `most_used_lang` is call 10 times and the debug-prints in the function print the username etc?

Reply:
For me it looks as if you would add a constant where condition that is `qry.filter(None == lang)` .. e.g. None would be returned by `most_used_lang` ...